dockers/docker-ptf/Dockerfile.j2

```diff
@@ -117,10 +131,88 @@ RUN rm -rf /debs \
     && pip install pybrctl pyro4 rpyc yabgp \
     && pip install unittest-xml-reporting \
     && pip install pyrasite \
+    && pip install retrying \
     && mkdir -p /opt       \
     && cd /opt             \
     && wget https://raw.githubusercontent.com/p4lang/ptf/master/ptf_nn/ptf_nn_agent.py
 
+RUN python3 -m venv env-python3
+
+# Activating a virtualenv. The virtualenv automatically works for RUN, ENV and CMD.
+ENV VIRTUAL_ENV=/root/env-python3
+ARG BACKUP_OF_PATH="$PATH"
+ENV PATH="$VIRTUAL_ENV/bin:$PATH"
+
+ENV LANG=C.UTF-8 LC_ALL=C.UTF-8 PYTHONIOENCODING=UTF-8
+
+RUN python3 -m pip install --upgrade --ignore-installed pip
+
+# Install all python modules from pypi. python3-scapy is exception, ptf debian package requires python3-scapy
+RUN python3 -m pip install setuptools \
+    && pip3 install supervisor \
+    && pip3 install ipython==5.4.1 \
+    && pip3 install Cython \
+    && git clone https://github.com/sflow/sflowtool \
```

Comment:
python venv should be python, not sure if sflowtool, openbfdd, nanomsg has python part?